lib/user-db.js

```diff
@@ -1,5 +1,6 @@
 const db = require('../db'),
-      async = require('async');
+      async = require('async'),
+      DB_NAME_BLACKLIST = /[^a-z0-9_\$\(\)\+\-/]/g;
```

Comment:
I'm terrible with regexes but I'm struggling reading this one.  I don't think special characters need escaping inside square brackets.

Also, where does this blacklist come from?  Is this a couch restriction, or something we've decided on ourselves?  Might be nice to document that somehow, e.g. a link to relevant couch docs from this code.